src/satbias/GsiSatBiasReader.cpp

```diff
@@ -0,0 +1,159 @@
+/*
+ * (C) Copyright 2021 UCAR
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+#include "GsiSatBiasReader.h"
+
+#include <string>
+#include <vector>
+#include <fstream>
+
+#include <Eigen/Dense>
+
+/// GSI predictor names
+/// Default predictors; some sensors use different predictors, see scanpos_predictors
+const std::vector<std::string> default_predictors = {"constant",
+                                                     "zenith_angle",
+                                                     "cloud_liquid_water",
+                                                     "lapse_rate_order_2",
+                                                     "lapse_rate",
+                                                     "cosine_of_latitude_times_orbit_node",
+                                                     "sine_of_latitude",
+                                                     "emissivity",
+                                                     "scan_angle_order_4",
+                                                     "scan_angle_order_3",
+                                                     "scan_angle_order_2",
+                                                     "scan_angle"
+                                                    };
+
+/// Predictors for
+/// goessndr .or. goes_img .or. ahi .or. seviri .or. ssmi .or. ssmis .or. gmi .or. abi
+/// (scan position instead of scan angle)
+const std::vector<std::string> scanpos_predictors = {"constant",
+                                                     "zenith_angle",
+                                                     "cloud_liquid_water",
+                                                     "lapse_rate_order_2",
+                                                     "lapse_rate",
+                                                     "cosine_of_latitude_times_orbit_node",
+                                                     "sine_of_latitude",
+                                                     "emissivity",
+                                                     "scan_position_order_4",
+                                                     "scan_position_order_3",
+                                                     "scan_position_order_2",
+                                                     "scan_position"
+                                                    };
```

Comment:
Just wondering if this is the best possible way and for future flexibility.
Would it be better to provide this information via the yaml file?

Reply:
@shlyaeva Yes. That is what I was thinking. One could put the list of predictors in the yaml.
For e.g. `satbias_converter.yaml` could be something like:
```
input file: testinput/satbias_crtm_in
default_predictors: &default_preds
  constant, zenith_angle, cloud_liquid_water, lapse_rate_order_2, lapse_rate, cosine_of_latitude_times_orbit_node, sine_of_latitude, emissivity, scan_angle_order_4,scan_angle_order_3, scan_angle_order_2, scan_angle
scanpos_predictors: &scanpos_preds
  constant, zenith_angle, cloud_liquid_water, lapse_rate_order_2, lapse_rate, cosine_of_latitude_times_orbit_node, sine_of_latitude, emissivity, scan_position_order_4, scan_position_order_3, scan_position_order_2, scan_position
output:
- sensor: amsua_n15
  output file: satbias_amsua_n15.nc4
  predictors: *default_preds
```

Thoughts?

Reply:
I think it's a very good idea; I'm working on it.

Reply:
I've changed to read predictor names from the yaml.